mqtt/mqtt-bridge/src/persist/storage/ring_buffer/block.rs

```diff
@@ -0,0 +1,280 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::hash_map::DefaultHasher,
+    fmt::Debug,
+    hash::{Hash, Hasher},
+};
+use crate::persist::storage::{
+    serialize::{binary_serialize, binary_serialize_size},
+    StorageResult,
+};
+use super::error::{BlockError, RingBufferError};
+
+/// A utility fn to calculate any entity that derives Hash.
+#[allow(dead_code)]
+pub(crate) fn calculate_hash<T>(t: &T) -> u64
+where
+    T: Hash + ?Sized,
+{
+    let mut hasher = DefaultHasher::new();
```

Comment:
It makes sense what you're saying. But before we jump to optimization we need to consider possible attack vectors. If we are sure that payload cannot be fully controlled by an adversary, we can use unsafe and fast hashing.

I did a quick search, and found this interesting SO post - https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed. There are some funny collisions for some algos :D

Reply:
I tried to write things in a way so that it would be fairly easy to change implementation. As I put things in a function to calculate hash, so we can modify as need be.

I agree the hashing algorithm that is default in rust is not the fastest, though I think we can stick with it for now and if we need to tune performance and validate security concerns on different hash algorithm we can :)